src/Sequence.cpp

```diff
@@ -308,4 +324,49 @@ Sequence::createCommandBuffer()
     KP_LOG_DEBUG("Kompute Sequence Command Buffer Created");
 }
 
+void
+Sequence::createTimestampQueryPool(uint32_t query_size)
+{
+    KP_LOG_DEBUG("Kompute Sequence creating query pool");
+    if (!this->mDevice) {
+        throw std::runtime_error("Kompute Sequence device is null");
+    }
+    if (!this->mPhysicalDevice) {
+        throw std::runtime_error("Kompute Sequence physical device is null");
+    }
+
+    vk::PhysicalDeviceProperties physicalDeviceProperties =
+      this->mPhysicalDevice->getProperties();
+    
+    if(physicalDeviceProperties.limits.timestampComputeAndGraphics){
+      vk::QueryPoolCreateInfo queryPoolInfo;
+      queryPoolInfo.setQueryCount(query_size);
+      queryPoolInfo.setQueryType(vk::QueryType::eTimestamp);
+      this->timestampQueryPool = std::make_shared<vk::QueryPool>(this->mDevice->createQueryPool(queryPoolInfo));
+
+      KP_LOG_DEBUG("Query pool for timestamps created");
+    }
+    else{
+      KP_LOG_DEBUG("Device does not support timestamps");
+    }
+}
+
+std::vector<std::uint64_t>
+Sequence::getTimestamps(){
+    if(!this->timestampQueryPool)
+        throw std::runtime_error("Timestamp latching not enabled");
+    
+    const auto n = this->mOperations.size()+1;
+    std::vector<std::uint64_t> timestamps(n, 0);
+    //XXX: the C++ method this->mDevice->getQueryPoolResults does not compile for me
```

Comment:
Is there a reason why getQueryPoolResutls doesn't compile? Could you share the error? We could open an issue in vkcpp, and I can try locally

Reply:
For consistency ideally we use the vk hpp header convention, if you look inside the header file of vkhpp do you see any differences? It normally just calls vk<func> underneath

Reply:
Works now